opendatahub-io/odh-dashboard
# Update 'Resource name' fields to meet UX guidelines: Serving Runtimes

Towards [RHOAIENG-14746](https://issues.redhat.com/browse/RHOAIENG-14746)

## Description
Updates the create/edit dialog for Model Servers to use the common `K8sNameDescriptionField` component in order to meet current UX guidelines.

## Screen shots

### Create
![image](https://github.com/user-attachments/assets/21027043-35cd-453c-b290-55433a95b015)

### Edit resource name popover
![image](https://github.com/user-attachments/assets/579a10c1-df07-4843-8720-7c42fc960142)

### Create editing Resource name
![image](https://github.com/user-attachments/assets/c9850f9f-5282-4e84-b8b6-45f6316fbba0)

### Edit
![image](https://github.com/user-attachments/assets/e3cbfbcb-1580-4d33-af37-cb1bc443241d)

## How Has This Been Tested?
- Navigate to a project this model servers and select the `Models` tab
  - Select `Add model server`
  - Enter the name for the new model server.
  - View the resource name via the `Edit resource name` link.
  - Change the resource name to something invalid (ie. contains a space)
  - Verify the submit button is disabled
  - Change the resource name to something valid
  - Verify the submit button is enabled
  - Save the profile.
  - Verify the profile is created with the correct Display name and Resource Name.
- Navigate to a project this model servers and select the `Models` tab
  - Select the kebab for an existing model server and choose `Edit`
  - Update the Name field
  - Notice the resource name is not editable
  - Save the model server.
  - Verify the model server is updated with the correct Display name.

## Test Impact
Updated unit tests for new fields

## Request review criteria:

/cc @simrandhaliw
